### PR TITLE
Qol 7312 move solr to ebs

### PR DESCRIPTION
**Pull request by ThrawnCA**
Keeping the index on EBS should have less file locking/consistency problems, is cheaper, and is a step toward being able to replicate and fail over gracefully (can't share the same EFS files between multiple instances, they need an exclusive lock).

### CODE REVIEW

File: recipes/solr-configure.rb [hunk 1]
```diff
@@ -56,7 +56,7 @@
 	action :touch
 end
 
-cookbook_file "/data/solr/log4j.properties" do
+cookbook_file "/var/solr/log4j.properties" do
```
Comment (duttonw) [inline recipes/solr-configure.rb:59]: new versions is now log4j2.xml

Comment (duttonw) [reply recipes/solr-configure.rb:59]: should we version control the new log4j so we can alter it?

**Review by duttonw — APPROVED** [review]

File: recipes/solr-configure.rb (hunk 1 above)
Comment (ThrawnCA) [reply recipes/solr-configure.rb:59]: Good point. I'll review and merge them on the other branch.

**Review by duttonw — COMMENTED** [review]
I want to see a cron job to backup solr, as well as something doing snapshots for us. 
and a script to restore solr.
i'm hesitant on moving it onto an ebs as opsworks does kill them if the machine is changed.

we may need to look at doing a solr cloud with 3 instances (both having solr and zookeeper installed) to manage failovers.

**Comment by ThrawnCA** [comment]
> I want to see a cron job to backup solr, as well as something doing snapshots for us.
> and a script to restore solr.

Fair point, we need something to propagate changes back to the EFS so that we can tear down and replace instances. Will do.

> we may need to look at doing a solr cloud with 3 instances (both having solr and zookeeper installed) to manage failovers.

That's a separate issue (QOL-4841) but I have plans for how we can do it without ZooKeeper. Besides the extra cost of a third instance, ZooKeeper would add a new single point of failure.

File: recipes/solr-configure.rb (hunk 1 above)
Comment (ThrawnCA) [reply recipes/solr-configure.rb:59]: Actually, having reviewed them, I don't think we need to customise the logging configuration at all. I'm not seeing anything in log4j.properties that we really need to keep.

**Comment by duttonw** [comment]
> > we may need to look at doing a solr cloud with 3 instances (both having solr and zookeeper installed) to manage failovers.
> 
> That's a separate issue (QOL-4841) but I have plans for how we can do it without ZooKeeper. Besides the extra cost of a third instance, ZooKeeper would add a new single point of failure.

zookeeper is also in tri mode so its not a single point of failure either, it might even be stronger. https://www.thegeekstuff.com/2016/10/zookeeper-cluster-install/

**Comment by duttonw** [comment]
Can you look at including the inbuilt backup functionality inside solr ontop of rsync as their own tools for backup should ensure no write logs or half data is copied.

https://lucene.apache.org/solr/guide/7_7/making-and-restoring-backups.html 
it can be configured to have the backup folder on the efs and then we have point in time backups as well as snapshot deltas (i believe)

**Comment by ThrawnCA** [comment]
> Can you look at including the inbuilt backup functionality inside solr ontop of rsync as their own tools for backup should ensure no write logs or half data is copied.

That does look very promising; thanks. Should be pretty straightforward to rsync a backed up copy instead of directly copying the index.

**Review by duttonw — APPROVED** [review]

**Pull request merged**